### PR TITLE
Fix leak of inline cache list when invalidating

### PR DESCRIPTION
**Pull request by curtisman**

### CODE REVIEW

**Comment by curtisman** [comment]
@kunalspathak @pleath Please code review.

**Comment by kunalspathak** [comment]
LGTM.

File: lib/Runtime/Base/ThreadContext.cpp [hunk 1]
```diff
@@ -2728,7 +2728,7 @@ ThreadContext::InvalidateInlineCacheList(InlineCacheList* inlineCacheList)
         }
     }
     NEXT_SLISTBASE_ENTRY;
-    inlineCacheList->Clear();
+    Adelete(&this->inlineCacheThreadInfoAllocator, inlineCacheList);
```
Comment (farzonl) [inline lib/Runtime/Base/ThreadContext.cpp]: why do you need to invalidate ever inlineCache in inlinecachelist if you are going to delete the inlinecachelist?

Comment (curtisman) [inline lib/Runtime/Base/ThreadContext.cpp]: The list is a list of pointer to inlineCache, deleting the list doesn't delete the inlineCache.
Zeroing the inlineCache will clear the now invalid information on it so tit can be used again.

**Comment by curtisman** [comment]
PR #1034 replaces this PR to retarget the change to release/1.2 branch

**Pull request closed without merging**